include/Server.hpp

```diff
@@ -0,0 +1,55 @@
+#ifndef SERVER_HPP
+# define SERVER_HPP
+# include <vector>
+# include <iostream>
+# include <sys/socket.h>
+# include <sys/types.h>
+# include <sys/event.h>
+# include <sys/time.h>
+# include <stdlib.h>
+# include <unistd.h>
+# include <netdb.h>
+# include <signal.h>
+# include "macro.hpp"
+# include "SockListen.hpp"
+
+class SockListen;
+class Server
+{
+	// private member
+	private:
+	std::string				_servername;
+	std::string				_root_dir;
+	std::vector<SockListen>	_listen_socks;
+
+	// base member function
+	protected:
+	Server();	// do not use : Server must be created with port(s), name, root_dir, main kqueue
+
+	public:
+	Server(std::vector<int> ports, std::string name, std::string root_dir, int kq);
+	~Server();
+	Server& operator=(const Server &source);
+	Server(const Server &source);
+
+	// getter
+	std::string				GetServername();
+	std::string				GetRootDir();
+	std::vector<SockListen>	*GetListeningSocks();
+	int						CheckIfListeningSock(int sock);
```

Comment:
Please, use lowerCaseCamelCase convention in writing functions and variables, variables always start. examples: _serverName; getServerName();
Also, please use socket instead of sock, because I can only think of socks when I see this word xD
Lastly, please don't use verbs in functions,  (if, at, with, etc...) functions need to be in Imperative format.

Reply:
Thanks for your review!

I'm changing the names, but about private members, I think that it is handy to use underscore in the beginning of word.
Like `int     _numPort;`
I will push again later, could you please check again?

Reply:
Modified and pushed!